### PR TITLE
Alt attribute for model-viewer element

### LINKED ISSUE #224: `<model-viewer>` can be configured with alt text
Also ensure that it is described correctly by screen readers when focused

This rolls into #187

### PR DESCRIPTION
**Pull request by cdata**
This change builds on #241 . Some details:

 - An `alt` attribute is introduced.
    - When configured, it will forward its value to the `<canvas>`'s `aria-label`
 - The `<canvas>` is now focusable by default
 - Default, generic `aria-label` text has been added to the `<canvas>` as a fallback.
 - The `<model-viewer>`'s shadow root is configured with [`delegatesFocus: true`](https://github.com/w3c/webcomponents/blob/gh-pages/proposals/ShadowRoot-delegatesFocus-Proposal.md#details).
    - Forwards its focus to its focusable shadow descendants (in `tabindex` order) when focused
    - Makes focus traversal less noisy, particularly when using screen readers

Here is an example of focus traversal with the AR button visible (in the recording I'm going back and forward across three focusable elements):

mvalttext

Fixes #224

### CODE REVIEW

File: src/template.js [hunk 1]
```diff
@@ -91,12 +91,12 @@ template.innerHTML = `
...
   </style>
   <div class="container">
-    <a class="enter-ar" href="#" aria-label="View this 3D model in augmented reality">
+    <a tabindex="2" class="enter-ar" href="#" aria-label="View this 3D model in augmented reality">
       <div class="disc"></div>
       ${ARGlyph}
     </a>
     <div class="poster"></div>
-    <canvas></canvas>
+    <canvas tabindex="1" aria-label="A canvas that displays a 3D model"></canvas>
```
[the remark this answers is not in the document]
Comment (cdata) [reply src/template.js]: VoiceOver reads 3D correctly. Will need to test other readers, but I don't have any installed right now.

We should discuss alternative defaults as a separate change, probably. Incorporating the URL or filename might work in some cases, but in others it is likely to be very noisy or difficult to derive (not all URLs point to files, or contain meaningful names).

[the remark this answers is not in the document]
Comment (cdata) [reply src/template.js]: For discovering screen reader pronunciations: https://github.com/GoogleWebComponents/model-viewer/issues/243

[the remark this answers is not in the document]
Comment (cdata) [reply src/template.js]: For discussing richer defaults: https://github.com/GoogleWebComponents/model-viewer/issues/244

[the remark this answers is not in the document]
Comment (cdata) [reply src/template.js]: I've changed the default `aria-label`

File: examples/augmented-reality.html [hunk 2]
```diff
@@ -58,6 +58,7 @@ <h1>&lt;model-viewer&gt; - Augmented Reality</h1>
             <h2>iOS, Magic Leap, WebXR</h2>
```
[the remark this answers is not in the document]
Comment (cdata) [reply examples/augmented-reality.html]: I added `alt` text to all of the `<model-viewer>` elements in all of the examples, as well as the glitch

File: src/model-viewer-base.js [hunk 3]
```diff
@@ -47,7 +48,7 @@ export const $renderer = Symbol('renderer');
  */
 export default class ModelViewerElementBase extends UpdatingElement {
   static get properties() {
-    return {src: {type: deserializeUrl}};
+    return {alt: {type: String}, src: {type: deserializeUrl}};
```
[the remark this answers is not in the document]
Comment (jsantell) [reply src/model-viewer-base.js]: Misread, sorry!

**Pull request merged**